src/ShopBundle/objects/TCMSTableEditor/TCMSShopTableEditor_ShopArticle.class.php

```diff
@@ -111,8 +114,30 @@ protected function PostSaveHook(&$oFields, &$oPostTable)
                 }
                 $this->oTableConf->SetLanguage($originalLanguageId);
             }
+
+            $stockMessage = $product->GetFieldShopStockMessage();
```

Comment:
- With this change, activation/deactivation is triggered in any case after saving. Instead it should only be triggered on actual stock changes.
- Logic for activation/deactivation depending on stock should not be duplicated. To change product stock, TShopArticle::UpdateStock() should be called, and any secondary effects like activating the product or its parent should be initiated from there.
- Stock updating is done in \ChameleonSystem\ShopBundle\ProductInventory\TableEditor, where UpdateStock is already called (I think activation/deactivation already works as expected when pressing Save within the stock iframe). That is where stock changes are actually performed, therefore I think changes in this file should be reverted.